torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -678,7 +678,9 @@ std::shared_ptr<JitFuture> TensorPipeAgent::send(
     {
       std::unique_lock<std::mutex> lock(timeoutMapMutex_);
       auto& timeoutFuturesVector = timeoutMap_[expirationTime];
-      timeoutFuturesVector.emplace_back(futureResponseMessage, timeout);
+      timeoutFuturesVector.emplace_back(
+          messageId, futureResponseMessage, timeout);
+      messageIdToExpiryMap_[messageId] = expirationTime;
```

Comment:
nit, feel free to ignore, if all we need is to pass the `expirationTime` to the response processing logic, we can potentially capture it in `pipeWrite` and then `pipeRead` to avoid global map insertion and lookup?